.github/workflows/test-cygwin.yml

```diff
@@ -0,0 +1,124 @@
+name: Test Cygwin
+
+on: [push, pull_request, workflow_dispatch]
+
+jobs:
+  build:
+    runs-on: windows-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-minor-version: [7, 8, 9]
+        architecture: ["x86", "x86_64"]
+
+    timeout-minutes: 40
+
+    name: Python 3.${{ matrix.python-minor-version }} ${{ matrix.architecture }}
+
+    steps:
+      - name: Fix line endings
+        run: |
+          git config --global core.autocrlf input
+
+      - name: Checkout Pillow
+        uses: actions/checkout@v3
+
+      - name: Install Cygwin
+        uses: cygwin/cygwin-install-action@v2
+        with:
+          platform: ${{ matrix.architecture }}
+          packages: >
+            ImageMagick gcc-g++ ghostscript jpeg libfreetype-devel
+            libimagequant-devel libjpeg-devel liblapack-devel
+            liblcms2-devel libopenjp2-devel libraqm-devel
+            libtiff-devel libwebp-devel libxcb-devel libxcb-xinerama0
+            make netpbm perl
+            python3${{ matrix.python-minor-version }}-cffi
+            python3${{ matrix.python-minor-version }}-cython
+            python3${{ matrix.python-minor-version }}-devel
+            python3${{ matrix.python-minor-version }}-numpy
+            python3${{ matrix.python-minor-version }}-sip
+            python3${{ matrix.python-minor-version }}-tkinter
+            qt5-devel-tools subversion xorg-server-extra zlib-devel
+
+      - name: Add Lapack to PATH
+        uses: egor-tensin/cleanup-path@v1
+        with:
+          dirs: 'C:\cygwin\bin;C:\cygwin\lib\lapack'
+          
+      - name: pip cache
+        uses: actions/cache@v3
+        with:
+          path: 'C:\cygwin\home\runneradmin\.cache\pip'
+          key: ${{ runner.os }}-cygwin-${{ matrix.architecture }}-pip3.${{ matrix.python-minor-version }}-${{ hashFiles('.ci/install.sh') }}
+          restore-keys: |
+            ${{ runner.os }}-cygwin-${{ matrix.architecture }}-pip3.${{ matrix.python-minor-version }}-
+
+      - name: Build system information
+        shell: bash.exe --login -eo pipefail "{0}"
+        run: |
+          python3 .github/workflows/system-info.py
+
+      - name: Install dependencies
+        run: |
+          dash.exe -c "python3.${{ matrix.python-minor-version }} -m pip install -U pip"
+          bash.exe .ci/install.sh
+
+      - name: Install a different NumPy
+        if: matrix.architecture == 'x86_64'
+        shell: dash.exe -l "{0}"
+        run: |
+          python3.${{ matrix.python-minor-version }} -m pip install -U 'numpy!=1.21.*'
```

Comment:
Simpler, but occasionally there's packaging problems pulling in a different python.  I have no idea how often this happens, since I just install all python versions available, but it's caused problems once on the NumPy CI in the six months to a year since they added a Cygwin run.  Are you okay with that risk?

Reply:
Ok, we can keep your version if you feel it is more stable.

Reply:
I changed the other references to `python3` to reflect that choice.  The other option is to tell alternatives to make sure it point python3 at the correct version of python, but I don't know how to do that.